# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1625575852 -7200
# Parent  86b9d4f1e2ca49d1446297e232edae64906d1b49
Branding 6: Clean up About box, switch off the updater, fix donation links.

diff --git a/mail/base/content/aboutDialog.xhtml b/mail/base/content/aboutDialog.xhtml
--- a/mail/base/content/aboutDialog.xhtml
+++ b/mail/base/content/aboutDialog.xhtml
@@ -1,10 +1,12 @@
 <?xml version="1.0"?> <!-- -*- Mode: HTML -*- -->
 
+#define BETTERBIRD 1
+
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 <?xml-stylesheet href="chrome://global/skin/global.css" type="text/css"?>
 <?xml-stylesheet href="chrome://messenger/content/aboutDialog.css" type="text/css"?>
 <?xml-stylesheet href="chrome://branding/content/aboutDialog.css" type="text/css"?>
 
@@ -34,19 +36,19 @@
 
   <script src="chrome://messenger/content/aboutDialog.js"/>
 #ifdef MOZ_UPDATER
   <script src="chrome://messenger/content/aboutDialog-appUpdater.js"/>
 #endif
 
   <vbox id="aboutDialogContainer">
     <hbox id="clientBox">
-      <vbox id="leftBox" flex="1"/>
+      <vbox id="leftBox" flex="0"/>
       <vbox id="rightBox" flex="1">
-        <hbox align="baseline">
+        <hbox align="baseline" style="margin-top: 10px;">
 #expand   <label id="version">__MOZ_APP_VERSION_DISPLAY__</label>
           <label is="text-link" id="releasenotes" hidden="true">&releaseNotes.link;</label>
         </hbox>
 
         <label id="distribution" class="text-blurb"/>
         <label id="distributionId" class="text-blurb"/>
 
         <vbox id="detailsBox">
@@ -108,45 +110,62 @@
           </vbox>
 
 #ifdef MOZ_UPDATER
           <description class="text-blurb" id="currentChannelText">
             &channel.description.start;<label id="currentChannel"/>&channel.description.end;
           </description>
 #endif
           <vbox id="experimental" hidden="true">
+#ifndef BETTERBIRD
+experimental is switched off via JS for release builds.
             <description class="text-blurb" id="warningDesc">
               &warningDesc.version;
 #ifdef MOZ_TELEMETRY_ON_BY_DEFAULT
               &warningDesc.telemetryDesc;
 #endif
             </description>
             <description class="text-blurb" id="communityExperimentalDesc">
               &community.exp.start;<label class="text-link" onclick="openLink('https://www.mozilla.org/');">&community.exp.mozillaLink;</label>&community.exp.middle;<label class="text-link" onclick="openAboutTab('about:credits');">&community.exp.creditsLink;</label>&community.exp.end;
             </description>
+#endif
           </vbox>
+#ifdef BETTERBIRD
+          <description class="text-blurb" style="margin-bottom: 0; font-size: 2em;">
+            <label class="text-link" onclick="openLink('https://www.betterbird.eu/');">www.betterbird.eu</label>
+          </description>
+#endif
           <description class="text-blurb" id="communityDesc">
+#ifndef BETTERBIRD
+communityDesc is switched off via JS for Daily builds.
             &community.start2;<label class="text-link" onclick="openLink('https://www.mozilla.org/');">&community.mozillaLink;</label>&community.middle2;<label class="text-link" onclick="openAboutTab('about:credits');">&community.creditsLink;</label>&community.end3;
+#endif
           </description>
           <description class="text-blurb" id="contributeDesc">
+#ifdef BETTERBIRD
+            &helpus.start;<label class="text-link" onclick="openLink('https://www.betterbird.eu/donate/');">&helpus.donateLink;</label>&helpus.end;
+#else
             &helpus.start;<label class="text-link" onclick="openLink('https://donate.mozilla.org/thunderbird/?utm_source=thunderbird-client&amp;utm_medium=referral&amp;utm_content=about-dialog');">&helpus.donateLink;</label>&helpus.middle;<label class="text-link" onclick="openLink('https://www.thunderbird.net/get-involved/')">&helpus.getInvolvedLink;</label>&helpus.end;
+#endif
           </description>
         </vbox>
       </vbox>
     </hbox>
+#ifndef BETTERBIRD
     <vbox id="bottomBox">
       <hbox pack="center">
         <label class="text-link bottom-link" onclick="openAboutTab('about:license');">&bottomLinks.license;</label>
         <label class="text-link bottom-link" onclick="openAboutTab('about:rights');">&bottomLinks.rights;</label>
         <label class="text-link bottom-link" onclick="openLink('https://www.mozilla.org/privacy/thunderbird/');">&bottomLinks.privacy;</label>
       </hbox>
       <description id="trademark">
         <label class="trademark-label">&trademarkInfo.part1;</label>
       </description>
     </vbox>
+#endif
   </vbox>
 
   <keyset id="mainKeyset">
     <key keycode="VK_ESCAPE" oncommand="window.close();"/>
 #ifdef XP_MACOSX
     <key id="key_close" modifiers="accel" key="&cmdCloseMac.commandKey;"
          oncommand="window.close();"/>
 #endif
diff --git a/mail/base/content/helpMenu.inc.xhtml b/mail/base/content/helpMenu.inc.xhtml
--- a/mail/base/content/helpMenu.inc.xhtml
+++ b/mail/base/content/helpMenu.inc.xhtml
@@ -1,39 +1,45 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+#define BETTERBIRD 1
+
 <menu id="helpMenu"
 #ifdef XP_WIN
             label="&helpMenuWin.label;" accesskey="&helpMenuWin.accesskey;"
 #else
             label="&helpMenu.label;" accesskey="&helpMenu.accesskey;"
 #endif
             onpopupshowing="buildHelpMenu();">
         <menupopup id="menu_HelpPopup">
+#ifndef BETTERBIRD
           <menuitem id="menu_openHelp"
                     label="&productHelp.label;"
 #ifndef XP_MACOSX
                     accesskey="&productHelp.accesskey;"
 #endif
                     key="key_openHelp"
                     oncommand="openSupportURL();"/>
           <menuitem id="menu_openTour"
                     label="&helpShowTour2.label;"
                     accesskey="&helpShowTour2.accesskey;"
                     oncommand="openLinkText(event, 'tourURL');"/>
+#endif
           <menuitem id="menu_keyboardShortcuts"
                     label="&helpKeyboardShortcuts.label;"
                     accesskey="&helpKeyboardShortcuts.accesskey;"
                     oncommand="openLinkText(event, 'keyboardShortcutsURL');"/>
+#ifndef BETTERBIRD
           <menuitem id="getInvolved"
                     label="&helpGetInvolvedPage.label;"
                     accesskey="&helpGetInvolvedPage.accesskey;"
                     oncommand="openLinkText(event, 'getInvolvedURL');"/>
+#endif
           <menuitem id="donationsPage"
                     label="&helpDonationsPage.label;"
                     accesskey="&helpDonationsPage.accesskey;"
                     oncommand="openLinkText(event, 'donateURL');"/>
           <menuitem id="feedbackPage" hidden="true"
                     label="&helpFeedbackPage.label;"
                     accesskey="&helpFeedbackPage.accesskey;"
                     oncommand="openFormattedURL('app.feedback.baseURL');"/>
diff --git a/mail/base/content/utilityOverlay.js b/mail/base/content/utilityOverlay.js
--- a/mail/base/content/utilityOverlay.js
+++ b/mail/base/content/utilityOverlay.js
@@ -206,20 +206,17 @@ function openLinkText(event, what) {
   switch (what) {
     case "getInvolvedURL":
       openUILink("https://www.thunderbird.net/get-involved/", event);
       break;
     case "keyboardShortcutsURL":
       openUILink("https://support.mozilla.org/kb/keyboard-shortcuts/", event);
       break;
     case "donateURL":
-      openUILink(
-        "https://donate.mozilla.org/thunderbird/?utm_source=thunderbird-client&utm_medium=referral&utm_content=help-menu",
-        event
-      );
+      openUILink("https://www.betterbird.eu/donate/", event);
       break;
     case "tourURL":
       openUILink("https://www.thunderbird.net/features/", event);
       break;
   }
 }
 
 /**
diff --git a/mail/branding/betterbird/content/aboutDialog.css b/mail/branding/betterbird/content/aboutDialog.css
--- a/mail/branding/betterbird/content/aboutDialog.css
+++ b/mail/branding/betterbird/content/aboutDialog.css
@@ -4,23 +4,23 @@
 }
 
 #leftBox {
   background-image: url("chrome://branding/content/about-logo.svg");
   background-repeat: no-repeat;
   /* min-width and min-height create room for the logo */
   min-width: 210px;
   min-height: 210px;
-  margin-top:20px;
   -moz-margin-start: 30px;
 }
 
 #rightBox {
   margin-left: 30px;
   margin-right: 30px;
+  margin-top: 40px;
 }
 
 #updateDeck > hbox > label:not([class="text-link"]) {
   color: #909090;
 }
 
 #trademark {
   font-size: xx-small;
diff --git a/mail/components/about-support/content/aboutSupport.js b/mail/components/about-support/content/aboutSupport.js
--- a/mail/components/about-support/content/aboutSupport.js
+++ b/mail/components/about-support/content/aboutSupport.js
@@ -103,16 +103,17 @@ function toFluentID(str) {
     .toLowerCase();
 }
 
 // Each property in this object corresponds to a property in Troubleshoot.jsm's
 // snapshot data.  Each function is passed its property's corresponding data,
 // and it's the function's job to update the page with it.
 var snapshotFormatters = {
   async application(data) {
+    // This comes form Services.appinfo.name likely via MOZ_APP_BASENAME.
     $("application-box").textContent = "Betterbird";
     $("useragent-box").textContent = data.userAgent;
     $("os-box").textContent = data.osVersion;
     if (AppConstants.platform == "macosx") {
       $("rosetta-box").textContent = data.rosetta;
     }
     $("binary-box").textContent = Services.dirsvc.get(
       "XREExeF",
diff --git a/mail/components/preferences/preferences.xhtml b/mail/components/preferences/preferences.xhtml
--- a/mail/components/preferences/preferences.xhtml
+++ b/mail/components/preferences/preferences.xhtml
@@ -60,16 +60,23 @@
   <link rel="localization" href="messenger/preferences/attachment-reminder.ftl" />
   <link rel="localization" href="messenger/preferences/passwordManager.ftl" />
   <link rel="localization" href="security/certificates/certManager.ftl" />
   <link rel="localization" href="security/certificates/deviceManager.ftl" />
 
   <script defer="defer" src="chrome://global/content/preferencesBindings.js"></script>
 #ifdef MOZ_UPDATER
   <script defer="defer" src="chrome://messenger/content/aboutDialog-appUpdater.js"></script>
+#else
+  <script defer="defer">
+    var { Services } = ChromeUtils.import("resource://gre/modules/Services.jsm");
+    var { XPCOMUtils } = ChromeUtils.import(
+      "resource://gre/modules/XPCOMUtils.jsm"
+    );
+  </script>
 #endif
   <script defer="defer" src="chrome://global/content/globalOverlay.js"></script>
   <script defer="defer" src="chrome://global/content/editMenuOverlay.js"></script>
   <script defer="defer" src="chrome://messenger/content/accountUtils.js"></script>
 
   <script defer="defer" src="chrome://communicator/content/contentAreaClick.js"></script>
   <script defer="defer" src="chrome://messenger/content/preferences/preferences.js"></script>
   <script defer="defer" src="chrome://messenger/content/preferences/findInPage.js"></script>
diff --git a/mail/confvars.sh b/mail/confvars.sh
--- a/mail/confvars.sh
+++ b/mail/confvars.sh
@@ -1,15 +1,17 @@
 #! /bin/sh
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
+# We need to keep MOZ_APP_NAME thunderbird, otherwise all the code that checks
+# AppConstants.MOZ_APP_NAME won't work any more, and there is such code in M-C!
 MOZ_APP_NAME=thunderbird
-MOZ_UPDATER=1
+# MOZ_UPDATER=1
 
 if test "$OS_ARCH" = "WINNT"; then
   if ! test "$HAVE_64BIT_BUILD"; then
     MOZ_VERIFY_MAR_SIGNATURE=1
   fi
 fi
 
 BROWSER_CHROME_URL=chrome://messenger/content/extensionPopup.xhtml
diff --git a/mail/moz.configure b/mail/moz.configure
--- a/mail/moz.configure
+++ b/mail/moz.configure
@@ -2,16 +2,17 @@
 # vim: set filetype=python:
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 set_config("MOZ_THUNDERBIRD", True)
 set_define("MOZ_THUNDERBIRD", True)
 
+# We can't set this to Betterbird or else Thunderbird profiles won't be recognised.
 imply_option("MOZ_APP_BASENAME", "Thunderbird")
 set_config("MOZ_APPUPDATE_HOST", "aus.thunderbird.net")
 imply_option("MOZ_CRASHREPORTER_URL", "https://crash-reports.thunderbird.net/")
 
 imply_option("--enable-default-browser-agent", False)
 
 
 @depends(target_is_windows, target_is_linux)
